### PR TITLE
feat(manager/flux): support system manifest inside subdirectories

### LINKED ISSUE #18517: [manager/flux] Support overriding `systemManifestRegex`
### What would you like Renovate to be able to do?

All my clusters define their system manifests in `{cluster-name}/flux-system/gitops-toolkit/gotk-components.yaml`.

Today, Renovate hardcodes [the regex that defines where this file is to be sought](https://github.com/renovatebot/renovate/blob/dca3418bbdbf1024ac6807e591562b4042bc846d/lib/modules/manager/flux/common.ts#L3) (i.e. `'(^|/)flux-system/gotk-components\\.yaml$'`). This means I can't adapt the manager to identify my own file as a system manifest.

I can of course [specify `fileMatch` on the manager](https://docs.renovatebot.com/modules/manager/flux/#file-matching) if I want to define additional places it should look for other manifests/resources, but, again, none of these will be treated as system manifests, and I thus cannot currently use Renovate to upgrade Flux.

### If you have any ideas on how this should be implemented, please tell us here.

A configuration option could be added to the Flux manager, though if I understand correctly, manager-specific config is currently not supported.

I don't know enough about Renovate's architecture or conventions to propose anything else beyond decoupling the current manager into two separate managers, e.g. "HelmRelease" (for `HelmRelease` resources only) and "FluxSystem" (exclusively for Flux system manifests).

### Is this a feature you are interested in implementing yourself?

Maybe

### PR DESCRIPTION
**Pull request by usmonster**
## Changes

Changed a regex so that the Flux manager will now detect system manifests in any subdirectory of `flux-system/`.

## Context

Closes #18517. (Supersedes #18523.)

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/flux/common.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { regEx } from '../../../util/regex';
 
-export const systemManifestRegex = '(^|/)flux-system/gotk-components\\.yaml$';
+export const systemManifestRegex = '(^|/)flux-system/(?:.+/)*gotk-components\\.yaml$';
```
Comment (viceice) [inline lib/modules/manager/flux/common.ts:3]:
```suggestion
export const systemManifestRegex = '(^|\\/)flux-system\\/(?:.+\\/)?gotk-components\\.yaml$';
```
that should be simpler and the same outcome

Comment (usmonster) [reply lib/modules/manager/flux/common.ts:3]: Thanks! Is the escaping needed here? I don't think the forward slash is special, and this makes it slightly less readable, but feel free to add the commit directly to the branch if you prefer!

Comment (viceice) [reply lib/modules/manager/flux/common.ts:3]: yes, as the slash is the start and end delimiter for a regex. the reason why it works without escape is that the js regex engine is not strict about that fact. so please be strict and properly escape.

Comment (usmonster) [reply lib/modules/manager/flux/common.ts]: Ah, I see your reasoning. Yes, the slash is special only in the context of regex literals, but there is no delimiter character at all in this context of a string pattern being passed to a regex constructor. I don't see it as a matter of engine strictness, but I applied your suggestion anyway. Thanks again for the review, and please let me know if you see any other issue with the PR!

File: lib/modules/manager/flux/common.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { regEx } from '../../../util/regex';
 
-export const systemManifestRegex = '(^|/)flux-system/gotk-components\\.yaml$';
+export const systemManifestRegex = '(^|\\/)flux-system\\/(?:.+\\/)?gotk-components\\.yaml$';
 export function isSystemManifest(file: string): boolean {
```
Comment (viceice) [inline lib/modules/manager/flux/common.ts:4]:
```suggestion

 export function isSystemManifest(file: string): boolean {
```
this probably needs a newline

**Review by viceice — COMMENTED** [review]
lint failure in `common.ts`

**Comment by viceice** [comment]
needs prettier run 😉

File: lib/modules/manager/flux/common.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { regEx } from '../../../util/regex';
 
-export const systemManifestRegex = '(^|/)flux-system/gotk-components\\.yaml$';
+export const systemManifestRegex =
+  '(^|\\/)flux-system\\/(?:.+\\/)?gotk-components\\.yaml$';
```
Comment (viceice) [inline lib/modules/manager/flux/common.ts:4]:
```suggestion
  '(^|\\/)flux-system\\/(?:.+\\/)?gotk-components\\.yaml$';

```

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**